coding/python_style_guide.rst

```diff
@@ -12,648 +13,671 @@ To request changes to these standards, please file an :ref:`RFC <decision-making
 
-Yes:
+.. _style-guide-py-files:
 
-.. code-block:: py
+7. Source Files & Modules
+=========================
 
-   if greeting:
+.. _style-guide-py-file-name:
 
-No:
+A Python source file name SHOULD be camelCase-with-leading-lowercase and ending in '.py'
+----------------------------------------------------------------------------------------
 
-.. code-block:: py
+A module containing a single class should be a ``camelCase``-with-leading-lowercase transliteration of the class's name.
 
-   if greeting == True:
-       pass
+The name of a test case should be descriptive without the need for a trailing numeral to distinguish one test case from another. 
 
-   if greeting is True:
-       pass
+.. TODO consider refactoring tests into their own section
 
-.. _style-guide-py-pitfalls:
+.. _style-guide-py-file-encoding:
 
-7. Programming Pitfalls
-=======================
+ASCII Encoding MUST be used for new code
+----------------------------------------
 
-.. _style-guide-py-7-1:
+Always use ASCII for new python code.
 
-``if x`` SHOULD NOT be used when you mean ``if x != None``
-----------------------------------------------------------
+- **Do not** include a coding comment (as described in  :pep:`263`) for ASCII files.
 
-Beware of writing ``if x`` when you mean ``if x != None``.
-This often comes up when testing whether a variable or argument that defaults to ``None`` was set to some other value.
-The other value might have a type (such as a container) that could be ``False`` in a boolean context!
+- Existing code using Latin-1 encoding (a.k.a. ISO-8859-1) is acceptable so long as it has a proper coding comment. All other code must be converted to ASCII or Latin-1 except for 3rd party packages used "as is."
 
-.. _style-guide-py-7-2:
+.. _style-guide-py-file-order:
 
-A mutable object MUST NOT be used as default in arg list
---------------------------------------------------------
+Standard code order SHOULD be followed
+--------------------------------------
 
-Never use a mutable object as default value in a function or method argument list.
-The problem is that the default value may itself change, leading to subtle bugs.
-This problem bites many new Python programmers, though usually only once.
-To avoid the problem use something like the following: 
+Within a module, follow the order: 
 
-.. code-block:: py
+1. Shebang line, ``#! /usr/bin/env python`` (only for executable scripts)
+2. Module-level comments (such as the `license statement <https://github.com/lsst/templates/blob/master/CopyrightHeader.py>`__)
+3. Module-level docstring
+4. Imports
+5. ``__all__`` statement, if any
+6. Private module variables (names start with underscore)
+7. Private module functions and classes (names start with underscore)
+8. Public module variables
+9. Public functions and classes
+10. Optional test suites
 
-   def proclist(alist=None):
-   if alist == None:
-   alist = []
+.. note:: Delete mention of test suites?
```

Comment:
Depends whether the word "suite" means `TestSuite` or if it means generically some test code. I think we need some clarification on what that means before deleting it so I think we have to keep it in for the moment and flag it with @ktlim for later removal.